src/device_trezor/device_trezor_base.cpp

```diff
@@ -275,22 +275,32 @@ namespace trezor {
       // Later if needed this generic message handler can be replaced by a pointer to
       // a protocol message handler which by default points to the device class which implements
       // the default handler.
+
+      if (m_last_msg_type == messages::MessageType_ButtonRequest){
+        on_button_pressed();
+      }
+      m_last_msg_type = input.m_type;
+
+      bool processed = true;
       switch(input.m_type){
         case messages::MessageType_ButtonRequest:
           on_button_request(input, dynamic_cast<const messages::common::ButtonRequest*>(input.m_msg.get()));
-          return true;
+          break;
         case messages::MessageType_PassphraseRequest:
           on_passphrase_request(input, dynamic_cast<const messages::common::PassphraseRequest*>(input.m_msg.get()));
-          return true;
+          break;
         case messages::MessageType_PassphraseStateRequest:
           on_passphrase_state_request(input, dynamic_cast<const messages::common::PassphraseStateRequest*>(input.m_msg.get()));
-          return true;
+          break;
         case messages::MessageType_PinMatrixRequest:
           on_pin_request(input, dynamic_cast<const messages::common::PinMatrixRequest*>(input.m_msg.get()));
-          return true;
+          break;
         default:
-          return false;
+          processed = false;
+          break;
       }
+
+      return processed;
```

Comment:
That seems to do the same thing as before. Is this intended ?

Reply:
I reverted it back now (during the refactoring I needed to do something after the processing, thus the return value. Then I realized I don't need that)